src/Microsoft.Identity.Web.OWIN/AppBuilderExtension.cs

```diff
@@ -175,6 +175,23 @@ public static IAppBuilder AddMicrosoftIdentityWebApp(
 
                                 if (clientInfoFromServer != null && clientInfoFromServer.UniqueTenantIdentifier != null && clientInfoFromServer.UniqueObjectIdentifier != null)
                                 {
+                                    var uniqueTenantIdentifierClaim = context.AuthenticationTicket.Identity.FindFirst(c => c.Type == ClaimConstants.UniqueTenantIdentifier);
+                                    var uniqueObjectIdentifierClaim = context.AuthenticationTicket.Identity.FindFirst(c => c.Type == ClaimConstants.UniqueObjectIdentifier);
+                                    if (uniqueTenantIdentifierClaim != null)
```

Comment:
It would be safer to compare `uniqueTenantIdentifierClaim` to `clientInfoFromServer.UniqueTenantIdentifier`

Reply:
Good observation! Fixed and tests added to check that